### ISSUE TITLE
 NOAUTH Authentication required after connecting to redis via redis:connect

### ISSUE THREAD

**Issue by mspanc** [dokku/dokku-redis#79]
I encounter `(error) NOAUTH Authentication required.` error when I am using redis:connect to get my redis CLI.

```
# dokku redis:connect myapp
127.0.0.1:6379> keys *
(error) NOAUTH Authentication required.
```

Dokku report:

```
# dokku report
-----> uname: Linux dokku-ovh1-003 4.4.0-59-generic #80-Ubuntu SMP Fri Jan 6 17:47:47 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
-----> memory:
              total        used        free      shared  buff/cache   available
Mem:          16037        5837         661         554        9538        9080
Swap:          1021         278         743
-----> docker version:
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64
 Experimental: false
-----> docker daemon info:
Containers: 66
 Running: 54
 Paused: 0
 Stopped: 12
Images: 523
Server Version: 17.05.0-ce
Storage Driver: aufs
 Root Dir: /home/docker-data/aufs
 Backing Filesystem: extfs
 Dirs: 656
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9048e5e50717ea4497b757314bad98ea3763c145
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-59-generic
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.66GiB
Name: dokku-ovh1-003
ID: ODSG:K5CP:S7PY:27AZ:T2LW:SHBK:OSEM:TT4X:C5LT:57HI:VJAI:FDQU
Docker Root Dir: /home/docker-data
Debug Mode (client): true
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
-----> sigil version: 0.4.0
-----> herokuish version:
herokuish: 0.3.29
buildpacks:
  heroku-buildpack-multi     v1.0.0
  heroku-buildpack-ruby      v159
  heroku-buildpack-nodejs    v99
  heroku-buildpack-clojure   v76
  heroku-buildpack-python    v99
  heroku-buildpack-java      v51
  heroku-buildpack-gradle    v21
  heroku-buildpack-grails    v21
  heroku-buildpack-scala     v76
  heroku-buildpack-play      v26
  heroku-buildpack-php       v121
  heroku-buildpack-go        v64
  heroku-buildpack-erlang    fa17af9
  buildpack-nginx            v8
-----> dokku version: 0.9.4
-----> dokku plugins:
 !     Deprecated: Please use plugin:list
plugn: 0.3.0
  00_dokku-standard    0.9.4 enabled    dokku core standard plugin
  20_events            0.9.4 enabled    dokku core events logging plugin
  apps                 0.9.4 enabled    dokku core apps plugin
  build-env            0.9.4 enabled    dokku core build-env plugin
  certs                0.9.4 enabled    dokku core certificate management plugin
  checks               0.9.4 enabled    dokku core checks plugin
  common               0.9.4 enabled    dokku core common plugin
  config               0.9.4 enabled    dokku core config plugin
  deployment-keys      0.1.0 enabled    Manage SSH deployment keys that should get injected into your containers on-build
  docker-options       0.9.4 enabled    dokku core docker-options plugin
  domains              0.9.4 enabled    dokku core domains plugin
  enter                0.9.4 enabled    dokku core enter plugin
  git                  0.9.4 enabled    dokku core git plugin
  hostkeys-plugin      0.1.0 enabled    Manage hostkeys (.ssh/known_hosts) in your container environment
  letsencrypt          0.8.6 enabled    Automated installation of let's encrypt TLS certificates
  logs                 0.9.4 enabled    dokku core logs plugin
  named-containers     0.9.4 enabled    dokku core named containers plugin
  nginx-vhosts         0.9.4 enabled    dokku core nginx-vhosts plugin
  plugin               0.9.4 enabled    dokku core plugin plugin
  postgres             1.0.0 enabled    dokku postgres service plugin
  proxy                0.9.4 enabled    dokku core proxy plugin
  ps                   0.9.4 enabled    dokku core ps plugin
  redis                1.0.0 enabled    dokku redis service plugin
  repo                 0.9.4 enabled    dokku core repo plugin
  shell                0.9.4 enabled    dokku core shell plugin
  slack                0.1.0 enabled    Notify Slack on deploy
  ssh-keys             0.9.4 enabled    dokku core ssh-keys plugin
  storage              0.9.4 enabled    dokku core storage plugin
  tags                 0.9.4 enabled    dokku core tags plugin
  tar                  0.9.4 enabled    dokku core tar plugin
```

**Comment by josegonzalez** [comment]
Does `redis-cli` allow you to specify the pass on connection via a flag?

**Comment by mspanc** [comment]
I don't have redis-cli on the host machine.

Inside the redis container:

```
root@1450529f5ef5:/data# redis-cli --version
redis-cli 3.2.5
```

```
# redis-cli --help
redis-cli 3.2.5

Usage: redis-cli [OPTIONS] [cmd [arg [arg ...]]]
  -h <hostname>      Server hostname (default: 127.0.0.1).
  -p <port>          Server port (default: 6379).
  -s <socket>        Server socket (overrides hostname and port).
  -a <password>      Password to use when connecting to the server.
  -r <repeat>        Execute specified command N times.
  -i <interval>      When -r is used, waits <interval> seconds per command.
                     It is possible to specify sub-second times like -i 0.1.
  -n <db>            Database number.
  -x                 Read last argument from STDIN.
  -d <delimiter>     Multi-bulk delimiter in for raw formatting (default: \n).
  -c                 Enable cluster mode (follow -ASK and -MOVED redirections).
  --raw              Use raw formatting for replies (default when STDOUT is
                     not a tty).
  --no-raw           Force formatted output even when STDOUT is not a tty.
  --csv              Output in CSV format.
  --stat             Print rolling stats about server: mem, clients, ...
  --latency          Enter a special mode continuously sampling latency.
  --latency-history  Like --latency but tracking latency changes over time.
                     Default time interval is 15 sec. Change it using -i.
  --latency-dist     Shows latency as a spectrum, requires xterm 256 colors.
                     Default time interval is 1 sec. Change it using -i.
  --lru-test <keys>  Simulate a cache workload with an 80-20 distribution.
  --slave            Simulate a slave showing commands received from the master.
  --rdb <filename>   Transfer an RDB dump from remote server to local file.
  --pipe             Transfer raw Redis protocol from stdin to server.
  --pipe-timeout <n> In --pipe mode, abort with error if after sending all data.
                     no reply is received within <n> seconds.
                     Default timeout: 30. Use 0 to wait forever.
  --bigkeys          Sample Redis keys looking for big keys.
  --scan             List all keys using the SCAN command.
  --pattern <pat>    Useful with --scan to specify a SCAN pattern.
  --intrinsic-latency <sec> Run a test to measure intrinsic system latency.
                     The test will run for the specified amount of seconds.
  --eval <file>      Send an EVAL command using the Lua script at <file>.
  --ldb              Used with --eval enable the Redis Lua debugger.
  --ldb-sync-mode    Like --ldb but uses the synchronous Lua debugger, in
                     this mode the server is blocked and script changes are
                     are not rolled back from the server memory.
  --help             Output this help and exit.
  --version          Output version and exit.

Examples:
  cat /etc/passwd | redis-cli -x set mypasswd
  redis-cli get mypasswd
  redis-cli -r 100 lpush mylist x
  redis-cli -r 100 -i 1 info | grep used_memory_human:
  redis-cli --eval myscript.lua key1 key2 , arg1 arg2 arg3
  redis-cli --scan --pattern '*:12345*'

  (Note: when using --eval the comma separates KEYS[] from ARGV[] items)

When no command is given, redis-cli starts in interactive mode.
Type "help" in interactive mode for information on available commands
and settings.
```

**Comment by josegonzalez** [comment]
This was fixed by #76, so upgrading the redis plugin should put you in a good place again.
